index.js

```diff
@@ -37,8 +37,11 @@ function compose (middleware) {
     function dispatch (i) {
       if (i <= index) return Promise.reject(new Error('next() called multiple times'))
       index = i
-      const fn = middleware[i] || next
+      let fn = middleware[i] || next
       if (!fn) return Promise.resolve()
+      if (fn._wrappers && Array.isArray(fn._wrappers) && fn._wrappers.length) {
+        fn = fn._wrappers.foreach((wrapper) => { fn = wrapper(fn) })
```

Comment:
I'm pretty sure this won't work, have you tested it? A lot of things seem off - I think you're confusing fn, the middleware being processed, with the middleware that compose returns.